### PR TITLE
[TRTLLM-11259][perf] Parallel VAE harness and implementation for WAN

### PR DESCRIPTION
**Pull request by NVShreyas**
Parallelizing VAE is important for scaling to multiple GPUs. VisionFly enables this by default and hence outperforms TRTLLM visual gen by upto 14% on 8 GPUs. This PR adds spatial-parallel VAE decoding using halo exchange, enabling the VAE decode step to run across all GPUs in the world. E2E latency reduction ranges from 4-9% across 2-8 gpus.

Implemented as a generic framework with a WAN-specific adapter, so future VAEs can be added by writing a thin adapter subclass.

performance - 

| Multi GPU Config | without parallel VAE | with parallel VAE |
|----------|----------|----------|
| CFG 2 UL 2        | 111.84 secs        | **107.42 secs** (-4%)        |
| CFG 2 UL 4        | 59.37 secs        | **54.83 secs** (-8%)        |

**This is enabled by default in WAN**

## Summary by CodeRabbit

* **New Features**
  * Added parallel VAE processing support for multi-GPU distributed video generation workflows
  * Added `--disable_parallel_vae` configuration flag to control parallel VAE usage in example scripts

* **Tests**
  * Added extensive multi-GPU test validation for parallel VAE, attention, convolution, and group normalization components

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/_torch/visual_gen/parallelism.py [hunk 1]
```diff
@@ -98,3 +98,33 @@ def setup_sequence_parallelism(
     model_config.ulysses_process_group = ulysses_pg
 
     return True, ulysses_size, ulysses_pg, ulysses_rank
+
+
+def setup_vae_parallelism(
```
Comment (zhenhuaw-me) [inline tensorrt_llm/_torch/visual_gen/parallelism.py]: It's not a big deal, but can we align the naming about "VAE Parallelism" vs "Parallel VAE"?

**Review by chang-l — COMMENTED** [review]
Nit (non-blocking for this PR): Would it make sense to structure this as a composite wrapper (similar to the our Ulysses attn), e.g.:
```
self.vae = ParallelVAEWan(vae_backend=self.vae, pg=pg)
```

File: tensorrt_llm/_torch/visual_gen/modules/attention.py [hunk 2]
```diff
@@ -295,3 +296,43 @@ def forward(
         out = self._attn_impl(q, k, v, batch_size, seq_len, kv_seq_len)
         out = self.to_out[0](out)
         return out
+
+
+class ParallelVaeAttentionBlock(torch.nn.Module):
```
[the remark this answers is not in the document]
Comment (NVShreyas) [reply tensorrt_llm/_torch/visual_gen/modules/attention.py]: yep

**Review by chang-l — APPROVED** [review]
